File: backend/app/models/enums.py
"""Enum types shared across models and schemas.

These inherit from str so they serialize to plain strings in JSON responses
and can be stored directly in SQLAlchemy String columns.
"""

from enum import Enum


class AccountType(str, Enum):
    CHECKING = "checking"
    SAVINGS = "savings"
    CREDIT_CARD = "credit_card"
    INVESTMENT = "investment"
    LOAN = "loan"


class TransactionType(str, Enum):
    INCOME = "income"
    EXPENSE = "expense"
    TRANSFER = "transfer"


class Frequency(str, Enum):
    WEEKLY = "weekly"
    BIWEEKLY = "biweekly"
    MONTHLY = "monthly"
    QUARTERLY = "quarterly"
    ANNUALLY = "annually"


# Groupings for net worth calculations
ASSET_ACCOUNT_TYPES = {AccountType.CHECKING, AccountType.SAVINGS, AccountType.INVESTMENT}
LIABILITY_ACCOUNT_TYPES = {AccountType.CREDIT_CARD, AccountType.LOAN}

File: frontend/src/types/enums.ts
export type AccountType = 'checking' | 'savings' | 'credit_card' | 'investment' | 'loan';
export type TransactionType = 'income' | 'expense' | 'transfer';
export type Frequency = 'weekly' | 'biweekly' | 'monthly' | 'quarterly' | 'annually';

export const ASSET_TYPES: AccountType[] = ['checking', 'savings', 'investment'];
export const LIABILITY_TYPES: AccountType[] = ['credit_card', 'loan'];

export const ACCOUNT_TYPE_OPTIONS: { value: AccountType; label: string }[] = [
  { value: 'checking', label: 'Checking' },
  { value: 'savings', label: 'Savings' },
  { value: 'credit_card', label: 'Credit Card' },
  { value: 'investment', label: 'Investment' },
  { value: 'loan', label: 'Loan' },
];

export const TRANSACTION_TYPE_OPTIONS: { value: TransactionType; label: string }[] = [
  { value: 'income', label: 'Income' },
  { value: 'expense', label: 'Expense' },
  { value: 'transfer', label: 'Transfer' },
];

export const FREQUENCY_OPTIONS: { value: Frequency; label: string }[] = [
  { value: 'weekly', label: 'Weekly' },
  { value: 'biweekly', label: 'Biweekly' },
  { value: 'monthly', label: 'Monthly' },
  { value: 'quarterly', label: 'Quarterly' },
  { value: 'annually', label: 'Annually' },
];
